TiFlash crash for invalid flag F in lock value

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->
ddl case
```
[2023/05/30 10:58:43.621 +08:00] [ERROR] [Exception.cpp:90] ["Code: 49, e.displayText() = DB::Exception: invalid flag F in lock value 533A74800000000000005C5F720380000000000000300380000000000000039080A0CE94ABE9900688AC01660621A5597584000C630621A5599BCC00046C0621A5597584000C0146, e.what() = DB::Exception, Stack trace:\n\n\n       0x1c206ce\tDB::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int) [tiflash+29492942]\n                \tdbms/src/Common/Exception.h:46\n       0x8076333\tDB::RecordKVFormat::decodeLockCfValue(DB::RecordKVFormat::DecodedLockCFValue&) [tiflash+134701875]\n                \tdbms/src/Storages/Transaction/TiKVRecordFormat.cpp:123\n       0x7feab1d\tDB::RegionCFDataBase<DB::RegionLockCFDataTrait>::insert(DB::StringObject<true>&&, DB::StringObject<false>&&, DB::DupCheck) [tiflash+134130461]\n                \tdbms/src/Storages/Transaction/RegionCFDataBase.cpp:57\n       0x7fe0015\tauto DB::Region::handleWriteRaftCmd(DB::WriteCmdsView const&, unsigned long, unsigned long, DB::TMTContext&)::$_14::operator()<unsigned long>(unsigned long) const [tiflash+134086677]\n                \tdbms/src/Storages/Transaction/Region.cpp:648\n       0x7fdef92\tDB::Region::handleWriteRaftCmd(DB::WriteCmdsView const&, unsigned long, unsigned long, DB::TMTContext&) [tiflash+134082450]\n                \tdbms/src/Storages/Transaction/Region.cpp:714\n       0x7f9d8ee\tDB::KVStore::handleWriteRaftCmd(DB::WriteCmdsView const&, unsigned long, unsigned long, unsigned long, DB::TMTContext&) const [tiflash+133814510]\n                \tdbms/src/Storages/Transaction/KVStore.cpp:297\n       0x7fbd075\tHandleWriteRaftCmd [tiflash+133943413]\n                \tdbms/src/Storages/Transaction/ProxyFFI.cpp:97\n  0x7f14a0ac7b98\t_$LT$engine_store_ffi..observer..TiFlashObserver$LT$T$C$ER$GT$$u20$as$u20$raftstore..coprocessor..QueryObserver$GT$::post_exec_query::h7baebf297f3c710d [[libtiflash_proxy.so](http://libtiflash_proxy.so/)+24021912]\n  0x7f14a19dec1d\traftstore::store::fsm::apply::ApplyDelegate$LT$EK$GT$::apply_raft_cmd::h25535f78b2e0a609 [[libtiflash_proxy.so](http://libtiflash_proxy.so/)+39844893]\n  0x7f14a19f8598\traftstore::store::fsm::apply::ApplyDelegate$LT$EK$GT$::process_raft_cmd::h61e8e50fe51b205f [[libtiflash_proxy.so](http://libtiflash_proxy.so/)+39949720]\n  0x7f14a19fdee4\traftstore::store::fsm::apply::ApplyDelegate$LT$EK$GT$::handle_raft_committed_entries::h5998616c7ccbd00a [[libtiflash_proxy.so](http://libtiflash_proxy.so/)+39972580]\n  0x7f14a19cf1cc\traftstore::store::fsm::apply::ApplyFsm$LT$EK$GT$::handle_apply::hdf9cbdff9eae6fb5 [[libtiflash_proxy.so](http://libtiflash_proxy.so/)+39780812]\n  0x7f14a19d3721\traftstore::store::fsm::apply::ApplyFsm$LT$EK$GT$::handle_tasks::hf5349c3a9c0f2237 [[libtiflash_proxy.so](http://libtiflash_proxy.so/)+39798561]\n  0x7f14a0bab98e\t_$LT$raftstore..store..fsm..apply..ApplyPoller$LT$EK$GT$$u20$as$u20$batch_system..batch..PollHandler$LT$raftstore..store..fsm..apply..ApplyFsm$LT$EK$GT$$C$raftstore..store..fsm..apply..ControlFsm$GT$$GT$::handle_normal::h95c5548f3d56678b [[libtiflash_proxy.so](http://libtiflash_proxy.so/)+24955278]\n  0x7f14a0b18393\tbatch_system::batch::Poller$LT$N$C$C$C$Handler$GT$::poll::h55f624a09f386174 [[libtiflash_proxy.so](http://libtiflash_proxy.so/)+24351635]\n  0x7f14a0c0fc18\tstd::sys_common::backtrace::__rust_begin_short_backtrace::h16cb33ef4f63f5a9 [[libtiflash_proxy.so](http://libtiflash_proxy.so/)+25365528]\n  0x7f14a0c596fe\tcore::ops::function::FnOnce::call_once$u7b$$u7b$vtable.shim$u7d$$u7d$::h3daa0229b294447f [[libtiflash_proxy.so](http://libtiflash_proxy.so/)+25667326]\n  0x7f14a219d2f5\tstd::sys::unix::thread::Thread::new::thread_start::hd2791a9cabec1fda [[libtiflash_proxy.so](http://libtiflash_proxy.so/)+47964917]\n                \t/rustc/96ddd32c4bfb1d78f0cd03eb068b1710a8cebeef/library/std/src/sys/unix/[thread.rs:108](http://thread.rs:108/)\n  0x7f149f0ae802\tstart_thread [libc.so.6+653314]\n  0x7f149f04e314\t__GI___clone [libc.so.6+258836]"] [source="DB::EngineStoreApplyRes DB::HandleWriteRaftCmd(const DB::EngineStoreServerWrap *, DB::WriteCmdsView, DB::RaftCmdHeader)"] [thread_id=102]
```
### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

### 4. What is your TiFlash version? (Required)
master
[2023/05/30 01:22:12.505 +08:00] [INFO] [client.go:509] ["Cluster version information"] [type=tikv] [version=7.2.0-alpha] [git_hash=a24d9d6838603b2b274376e52f92f67a10fecd84]
[2023/05/30 01:22:12.505 +08:00] [INFO] [client.go:509] ["Cluster version information"] [type=pd] [version=7.2.0-alpha] [git_hash=a432973740d5c80f3c80ec0de916937f805f971c]
[2023/05/30 01:22:12.505 +08:00] [INFO] [client.go:509] ["Cluster version information"] [type=tiflash] [version=7.2.0-alpha] [git_hash=f794c8f7fc19bd3fa04cbbf4e201d70bae0cb8fe]
[2023/05/30 01:22:12.505 +08:00] [INFO] [client.go:509] ["Cluster version information"] [type=tidb] [version=7.2.0-alpha] [git_hash=10e6e427303e02e0cb0101bb4bf9d8df32c36bc8]

--- a/dbms/src/Storages/Transaction/TiKVRecordFormat.cpp
+++ b/dbms/src/Storages/Transaction/TiKVRecordFormat.cpp
@@ -117,6 +117,11 @@ inline void decodeLockCfValue(DecodedLockCFValue & res)
                 UNUSED(txn_source_prefic);
                 break;
             }
+            case PESSIMISTIC_LOCK_WITH_CONFLICT_PREFIX:
+            {
+                // https://github.com/pingcap/tidb/issues/43540
+                break;
+            }
             default:
             {
                 std::string msg = std::string("invalid flag ") + flag + " in lock value " + value.toDebugString();

--- a/dbms/src/Storages/Transaction/TiKVRecordFormat.h
+++ b/dbms/src/Storages/Transaction/TiKVRecordFormat.h
@@ -68,6 +68,7 @@ static const char GC_FENCE_PREFIX = 'F';
 static const char LAST_CHANGE_PREFIX = 'l';
 static const char TXN_SOURCE_PREFIX_FOR_WRITE = 'S';
 static const char TXN_SOURCE_PREFIX_FOR_LOCK = 's';
+static const char PESSIMISTIC_LOCK_WITH_CONFLICT_PREFIX = 'F';
 
 static const size_t SHORT_VALUE_MAX_LEN = 64;
 

